src/samples/test-bots/TeamsSSOAgent/Program.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using Microsoft.Agents.Authentication;
+using Microsoft.Agents.Builder;
+using Microsoft.Agents.Builder.App;
+using Microsoft.Agents.Builder.App.UserAuth;
+using Microsoft.Agents.Extensions.Teams.App.UserAuth;
+using Microsoft.Agents.Hosting.AspNetCore;
+using Microsoft.Agents.Storage;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System.Threading;
+using TeamsSSOAgent;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddHttpClient();
+
+builder.Logging.AddConsole();
+builder.Logging.AddDebug();
+
+// Add AgentApplicationOptions
+builder.AddAgentApplicationOptions();
+
+// Add the Agent
+builder.AddAgent<AuthAgent>();
+
+// Register IStorage.  For development, MemoryStorage is suitable.
+// For production Agents, persisted storage should be used so
+// that state survives Agent restarts, and operate correctly
+// in a cluster of Agent instances.
+builder.Services.AddSingleton<IStorage, MemoryStorage>();
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+app.UseRouting();
```

Comment:
need routing?

Reply:
This is POC, Rido... not the final deal.  It is for comment on what it's doing.